CharlieBackend.Api.UnitTes/CurrentUserTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Moq;
+using Xunit;
+using System;
+using System.Security.Claims;
+using System.Security.Principal;
+using Microsoft.AspNetCore.Http;
+using CharlieBackend.Core.Entities;
+using CharlieBackend.Business.Services;
+using FluentAssertions;
+
+
+namespace CharlieBackend.Api.UnitTest
+{
+    public class CurrentUserTests : TestBase
+    {
+        private readonly Mock<IHttpContextAccessor> _httpContextAccessorMock;
+        private readonly DefaultHttpContext _httpContext;
+        private readonly Mock<GenericIdentity> _mockIdentity;
+        private  CurrentUserService _currentuserService;
+
+        public CurrentUserTests()
+        {
+            _httpContextAccessorMock = new Mock<IHttpContextAccessor>();
+            _httpContext = new DefaultHttpContext();
+            _mockIdentity = new Mock<GenericIdentity>("Test");
+            _currentuserService = new CurrentUserService(_httpContextAccessorMock.Object);
+        }
+
+        [Fact]
+        public void ThrowErrorIfNotLoggined()
+        {
+            _httpContextAccessorMock.Setup(req => req.HttpContext)
+                .Returns(new DefaultHttpContext());
+
+            Func<long> getId = () => _currentuserService.AccountId;
+            Action act = () => getId();
+
+            //Assert
+            act.Should().Throw<UnauthorizedAccessException>();
+        }
+
+        [Fact]
+        public void getAccountId()
+        {
+            _mockIdentity.Setup(i => i.Claims)
+                .Returns(new Claim[]{
+                    new Claim("AccountId", "123") 
+                });
+            _httpContext.User = new ClaimsPrincipal(_mockIdentity.Object);
+            _httpContextAccessorMock.Setup(req => req.HttpContext).Returns(_httpContext);
+
+            _currentuserService.AccountId.Should().Be(123);
```

Comment:
When introducing expected values as constants, what I would suggest is declaring them in a single place (at least in the `Arrange` section of the test). This way you're decreasing the possibility of making a mistake and making code more "fluent".
Something like this:

```cs
            // Arrange
            long expectedAccountId = 123;
            
            _mockIdentity.Setup(i => i.Claims)
                .Returns(new Claim[]{
                    new Claim("AccountId", expectedAccountId.ToString()) 
                });

            // Act & Assert
            _currentuserService.AccountId.Should().Be(expectedAccountId);
```

This comment also applies to other methods & constants of yours.

Reply:
Solved in https://github.com/ita-social-projects/WhatBackend/pull/504/commits/599f172a458febfc3f38a1fd66639d04ac39d823